dsgrid/config/dimensions.py

```diff
@@ -389,31 +395,81 @@ class DateTimeDimensionModel(TimeDimensionBaseModel):
         options=TimeZone.format_descriptions_for_docs(),
     )
 
-    @validator("ranges", pre=True)
-    def check_times(cls, ranges, values):
-        return _check_time_ranges(ranges, values["str_format"], values["frequency"])
+    @root_validator(pre=False)  # TODO: modify as model works with more time_type schema
+    def check_time_type(cls, values):
+        if (
+            values["class_name"] == "Time" and values["time_type"] == TimeDimensionType.DATETIME
+        ) or (
+            values["class_name"] == "AnnualTime"
+            and values["time_type"] == TimeDimensionType.ANNUAL
+        ):
+            pass
+        else:
+            raise ValueError(
+                f'time_type={values["time_type"].value} does not match class_name={values["class_name"]}. \n'
+                " * For class=Time, use time_type=datetime. \n"
+                " * For class=AnnualTime, use time_type=annual."
+            )
+        return values
 
-    @validator("frequency")
-    def check_frequency(cls, value):
-        if value == timedelta(days=366):
+    @root_validator(pre=False)
+    def check_frequency(cls, values):
+        if values["frequency"] in [timedelta(days=365), timedelta(days=366)]:
             raise ValueError(
-                "366 days not allowed for frequency, use 365 days to specify annual frequency."
+                f'frequency={values["frequency"]}, 365 or 366 days not allowed, '
+                "use class=AnnualTime, time_type=annual to specify a year series."
             )
-        return value
+        return values
+
+    @validator("ranges", pre=True)
+    def check_times(cls, ranges, values):
+        return _check_time_ranges(ranges, values["str_format"], values["frequency"])
 
 
 class AnnualTimeDimensionModel(TimeDimensionBaseModel):
     """Defines an annual time dimension where timestamps are years."""
 
+    time_type: TimeDimensionType = Field(default=TimeDimensionType.ANNUAL)
+    str_format: Optional[str] = Field(
+        title="str_format",
+        default="%Y",
+        description="Timestamp string format",
+        notes=(
+            "The string format is used to parse the timestamps provided in the time ranges."
+            "Cheatsheet reference: `<https://strftime.org/>`_.",
+        ),
+    )
+    ranges: List[TimeRangeModel] = Field(
+        title="time_ranges",
+        description="Defines the contiguous ranges of time in the data, inclusive of start and end time.",
+    )
     include_leap_day: bool = Field(
         title="include_leap_day",
         default=False,
-        description="""
-        Whether annual time includes leap day. Accepted: 
-            True, False
-        """,
+        description="Whether annual time includes leap day.",
     )
 
+    @root_validator(pre=False)  # TODO: modify as model works with more time_type schema
+    def check_time_type(cls, values):
+        if (
```

Comment:
Assuming this is the same logic as above, can you put it in a function and then call it from both places?